File: 07-recursive-quotient/recursive_quotient.py
def recursive_quotient(a: int, b: int) -> int:
    # Handle negative numbers
    if a < 0 and b < 0:
        return recursive_quotient(-a, -b)
    elif a < 0:
        return -recursive_quotient(-a, b)
    elif b < 0:
        return -recursive_quotient(a, -b)
    
    # Base case
    if a < b:
        return 0
    
    # Recursive case
    return 1 + recursive_quotient(a - b, b)

def main():
    # Read input
    with open('input.txt', 'r') as f:
        a, b = map(int, f.read().strip().split())
    
    # Calculate result
    result = recursive_quotient(a, b)
    
    # Write output
    with open('output.txt', 'w') as f:
        f.write(str(result) + '\n')

if __name__ == "__main__":
    main() 

File: 07-recursive-quotient/recursive_quotient.cpp
#include <iostream>
#include <fstream>
using namespace std;

int recursiveQuotient(int a, int b) {
    // Handle negative numbers
    if (a < 0 && b < 0) {
        return recursiveQuotient(-a, -b);
    }
    else if (a < 0) {
        return -recursiveQuotient(-a, b);
    }
    else if (b < 0) {
        return -recursiveQuotient(a, -b);
    }
    
    // Base case
    if (a < b) {
        return 0;
    }
    
    // Recursive case
    return 1 + recursiveQuotient(a - b, b);
}

int main() {
    // Read input
    std::ifstream input("input.txt");
    int a, b;
    input >> a >> b;
    input.close();
    
    // Calculate result
    int result = recursiveQuotient(a, b);
    
    // Write output
    ofstream outFile("output.txt");
    outFile << result << endl;
    
    return 0;
} 